implementations/micrometer-registry-prometheus/src/main/java/io/micrometer/prometheusmetrics/MicrometerCollector.java

```diff
@@ -44,12 +44,15 @@ class MicrometerCollector implements MultiCollector {
 
     private final String conventionName;
 
+    private final Meter.Id id;
+
     private final List<String> tagKeys;
 
     // take name to avoid calling NamingConvention#name after the call-site has already
     // done it
     MicrometerCollector(String name, Meter.Id id, NamingConvention convention) {
         this.conventionName = name;
+        this.id = id;
```

Comment:
If we only want this for the `Type`, maybe we should just store that instead of the `Id`? It's a bit confusing what a MicrometerCollector's `Id` is since it is a collector for potentially multiple IDs.